### PR TITLE
Mock crust

### PR DESCRIPTION
**Pull request by madadam**
Minimal version of mock crust. Currently simulates only idealized network conditions where all messages are always delivered. More functionality coming later, for example simulating various NAT types, or unstable network connections, etc...

[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/914)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @brian-js

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/core.rs [hunk 1]
```diff
@@ -800,6 +831,7 @@ impl Core {
     fn dispatch_request_response(&mut self,
                                  routing_msg: RoutingMessage)
                                  -> Result<(), RoutingError> {
+        trace!("{:?}Handling - {:?}", self, routing_msg);
```
Comment (afck) [inline src/core.rs]: I wouldn't reintroduce those trace statements: We log most of the requests and responses in the handling methods.
(Or was that a git rebase gone wrong?)

File: src/core.rs [hunk 2]
```diff
@@ -236,11 +270,7 @@ impl Core {
                                                         category_tx);
 
         // TODO(afck): Add the listening port to the Service constructor.
```
Comment (afck) [inline src/core.rs]: That comment now probably belongs in the closure in non-test `new`.

File: src/core.rs (hunk 1 above)
Comment (madadam) [inline src/core.rs]: Definitely a rebase hiccup :grin: Will remove it.

**Comment by madadam** [comment]
This PR is superseded by https://github.com/maidsafe/routing/pull/920, closing.

**Pull request closed without merging**